test/mocha/10-api.js

```diff
@@ -34,4 +36,168 @@ describe('bedrock-app-identity API', () => {
       'Ed25519VerificationKey2020');
     keys.keyAgreementKey.type.should.equal('X25519KeyAgreementKey2020');
   });
+  it('throws error when attempting to configure app identity again',
+    async () => {
+      let err;
+      let res;
+      try {
+        // this will trigger the bedrock.init event which will call
+        // `_generateApplicationIdentity` and try to configure app identity
+        // again
+        res = await bedrock.events.emit('bedrock.init');
```

Comment:
This does not seem like a safe thing to do. There are other modules that are running that would also receive this event -- including those that were used to run the test suite. We need to figure out another way to test this, IMO.

Reply:
@dlongley it looks like the only other module running  is `bedrock-test` and it's not listening to that event.
However another option i can think of is that we can export that callback function from `main.js` for testing purposes.  Does that seem okay or do you have any suggestions?

Reply:
@JSAssassin, I don't really like that option either because it would imply we've got to change code everywhere to test event handlers ... which would be a mess and would harm code readability. I think we need more opinions on this.

Reply:
I'm jumping into something I'm probably not qualified to comment on, but maybe we should export `_generateApplicationIdentity` and then test it outside of the event emitter?

Reply:
+1 to @aljones15's suggestion.